command/taint.go

```diff
@@ -105,43 +109,16 @@ func (c *TaintCommand) Run(args []string) int {
 		return 1
 	}
 
-	// Get the proper module we want to taint
-	modPath := strings.Split(module, ".")
-	mod := s.ModuleByPath(modPath)
-	if mod == nil {
-		if allowMissing {
-			return c.allowMissingExit(name, module)
-		}
-
-		c.Ui.Error(fmt.Sprintf(
-			"The module %s could not be found. There is nothing to taint.",
-			module))
-		return 1
-	}
-
-	// If there are no resources in this module, it is an error
-	if len(mod.Resources) == 0 {
-		if allowMissing {
-			return c.allowMissingExit(name, module)
-		}
+	// Get resource type from name
+	resourceType := strings.Split(name, ".")[0]
```

Comment:
If I'm understanding correctly, this information is available in `rsa.Type`, and so we shouldn't need to extract it and pass it around as a separate argument.